### ISSUE TITLE
Clarification of identity

### ISSUE THREAD

**Issue by leo-lb** [maidsafe/parsec#345]
Hello,
I am studying PARSEC here and there since a while but I remain confused about one thing.
To me, PARSEC is not byzantine fault tolerant because it has no mechanism to restrict one malicious actor from creating as many identities inside the gossip graph as necessary to take it over and own the consensus. (Sybil attack)
Is there something I am not seeing here?
Thank you

**Comment by jeanphilippeD** [comment]
Hello,

Thank you for your question.

New identities are only added to participate in the gossip, if consensus is reached to add them, It delegate to its user the choice of which new identities to add.

The user of Parsec (for us the Routing+Vault crate), implement a Sybil resistance mechanism in the choice of Nodes to Add and Remove from Parsec.

In Routing + Vault, we implement Sybil resistance through a Node Ageing process, where only the node that have done most work for the network participate in Parsec.

Kind Regards.

**Comment by leo-lb** [comment]
Thanks for your answer,

So it means that fully automated systems cannot be byzantine fault tolerant with PARSEC?

Node Ageing is only an additional step to perform in order to conduct a Sybil attack, it does not prevent it. Nodes can avoid byzantine faults for a while and then start misbehaving. I guess it is acceptable, but I think I could still attack the network in practice, and because there's no authority, no one can stop it effectively.

Do you agree with my conclusions or there is something I am still not getting?

**Comment by theWebalyst** [comment]
@leo-lb to understand this I think you need to understand more of the picture. To start with, I'd recommend https://primer.safenetwork.org/ and follow up with any questions at s as https://safenetforum.org 

TLDR; no, it is not easy to Sybil attack the network. The difficulty is lower while the network is small and increases as the network scales up. So if you want to attempt it, get in early!

**Comment by leo-lb** [comment]
@theWebalyst Thanks.

Sybil attack:
It is in the interest of an attacker (e.g. NSA) to follow emerging technologies such as the SAFE Network and get in early with lots of servers of their control to later influence the network.

**Comment by octol** [comment]
> So it means that fully automated systems cannot be byzantine fault tolerant with PARSEC?

Note though that Byzantine fault tolerance (BFT) is not the same as resistance to Sybil attacks (fake identities). PARSEC is indeed BFT in that it can tolerate a fraction of malicious nodes.

Protecting against Sybil attacks is a separate thing compared to BFT and as mentioned above, handled higher up in the stack by node ageing, by not allowing nodes to participate in voting unless they are vetted first.

**Comment by leo-lb** [comment]
> Protecting against Sybil attacks is a separate thing compared to BFT and as mentioned above, handled higher up in the stack by node ageing, by not allowing nodes to participate in voting unless they've been vetted first.

But the question is vetted by who? Whether that is, indeed, or not, included in the definition of BFT, I do not know. But to me it is an important concern, that 33% or even 80% of nodes remains trivial to create because identity creation has no cost, you do not even need much bandwidth or computing resources to achieve this. A single program that's optimized for this use case can handle hundreds of thousands of connections if necessary on a single machine in order to conduct the Sybil attack, no matter the size of the network.

Of course, I will wait until the first public MVP is out to conduct experiments on a testnet and rule out definitive conclusions but I am a bit worried about this problem.

I hope you will put great thinking into solving this Sybil attack issue.

Thanks everyone for your answers.

**Comment by theWebalyst** [comment]
@leo-lb you need to understand the measures which prevent the attack you have in mind, starter links in my earlier response. For example, it is not the case that you can create identities at little cost in order to mount a successful Sybil attack. It is in fact very costly.

**Comment by leo-lb** [comment]
@theWebalyst I am not convinced of this. I have already read the links you provided.

"Resource Proof - a test of bandwidth, storage space and CPU power. If a Vault trying to join the Network fails the test, it will not be allowed to join."

How do you ensure that:
- It is not discriminatory (e.g. low power IOT, low speed network in poor countries, ..)
- That it cannot be performed once and reused

Also, what are the technical details of this resource proof?

- How do you prove bandwidth in a way that's verifiable for the rest of the network?
- How do you prove storage space through time? (e.g. some storage space can be used to generate proof for Node A, then later that same storage space can be wiped and re-used to generate proof of Node B, therefore, it would not be useful to use it as part of the proof besides guaranteeing proofs can only be generated sequentially)
- How do you prove CPU power? (That should be OK)

The resistance to malicious attackers here is more so provided by this proof than the rest of PARSEC itself. It's Proof of Work just like Bitcoin. If you have computing resources then you own the network.
